src/ctia/domain/id.clj

```diff
@@ -0,0 +1,109 @@
+(ns ctia.domain.id
+  (:require [ctia.lib.url :as url])
+  (:import java.lang.String))
+
+(def short-id-re
+  (re-pattern (str "(" url/url-chars-re ")")))
+
+(def url-re
+  #"(https?):\/\/([-\da-zA-Z][-\da-zA-Z.]*)(:(\d+))?((\/[-\w.]+)*)\/ctia\/([a-z]+)\/")
+
+(def long-id-re
+  (re-pattern
+   (str url-re
+        short-id-re)))
+
+(defn make-long-id-str
+  [{:keys [hostname short-id path-prefix port protocol type]
+    :as parts}]
+  (when parts
+    (str protocol
+         "://"
+         hostname
+         (if port (str ":" port))
+         path-prefix
+         "/ctia/"
+         type
+         "/"
+         (url/encode short-id))))
+
+(defn long-id?
+  "Make an educated guess if this is a long-form ID"
+  [^String id]
+  ;; TODO - More specific checking?
+  (boolean (.startsWith id "http")))
```

Comment:
is it necessary to explicitly cast a java.lang.Boolean to a java.lang.Boolean? I suggest getting rid of it, and changing the defn to:
`(s/defn long-id? :- s/Bool [^String id] (.startsWith id "http"))`
(with docs, of course)